input/elasticapm/internal/decoder/stream_decoder_test.go

```diff
@@ -112,3 +112,33 @@ func TestNDStreamReaderReadAhead(t *testing.T) {
 		}
 	}
 }
+
+func TestNDStreamDecoderTrailing(t *testing.T) {
+	dec := NewNDJSONStreamDecoder(strings.NewReader(`{"key":"value1"}junk`+"\n"), 100)
+
+	var out struct{ Key string }
+	require.NoError(t, dec.Decode(&out))
```

Comment:
Shouldn't this be an error since the line is not a valid JSON? Correct me if I am wrong, but my understanding was that in NDJSON we will delimit by `\n` and each line should be a valid JSON.

Reply:
Yes that would be better. We just decode the first JSON object off the line, and ignore anything after it. This is pre-existing behaviour, so I'll defer improving on that.